### PR TITLE
Modify the login process to user the new Web Api

### PR DESCRIPTION
**Pull request by wizofe**
Create a wrapper for the kano-webengine as to use the new web api login and registration process.

### CODE REVIEW

**Comment by wizofe** [comment]
[(5) As an engineer, I want the new web-based registration flow to replace the old GTK one (i.e. set up the new project)](https://trello.com/c/hTp1IFfL/1887-5-as-an-engineer-i-want-the-new-web-based-registration-flow-to-replace-the-old-gtk-one-ie-set-up-the-new-project)

**Comment by wizofe** [comment]
[(5) As an engineer, I want the new web-based registration flow to replace the old GTK one (i.e. set up the new project)](https://trello.com/c/hTp1IFfL/1887-5-as-an-engineer-i-want-the-new-web-based-registration-flow-to-replace-the-old-gtk-one-ie-set-up-the-new-project)

File: bin/kano-login [hunk 1]
```diff
@@ -87,10 +88,10 @@ class MainWindow(ApplicationWindow):
         apply_styling_to_screen(css_path)
 
         if screen == "register":
-            # Gives the user the option to log in or register
             FirstScreen(self)
         elif screen == "login":
```
Comment (skarbat) [inline bin/kano-login]: Could we move the `login` logic into a class method? Feel a bit uncomfortable doing all this stuff in the constructor...

File: kano_login/first_screen.py [hunk 2]
```diff
@@ -66,6 +66,12 @@ def __init__(self):
 
 
 class FirstScreen(FirstScreenTemplate):
+
+    webengine = '/usr/bin/kano-webengine'
+    web_api = 'http://10.49.104.215:8080/'
```
Comment (skarbat) [inline kano_login/first_screen.py]: hardcoded ip and port... mmmm... could we live with a command line option, environment or something?

File: kano_login/first_screen.py [hunk 3]
```diff
@@ -66,6 +66,12 @@ def __init__(self):
...
+    signup = '#/signup'
+    win_size = '-W 590 -H 600'
```
Comment (skarbat) [inline kano_login/first_screen.py]: likewise for view geometry?

File: kano_login/first_screen.py [hunk 4]
```diff
@@ -93,18 +99,21 @@ def login_screen(self, widget, event):
...
+            os.system('{} {} {}'.format(FirstScreen.webengine, FirstScreen.web_api, FirstScreen.win_size))
         else:
             NoInternet(self.win)
 
     def register_screen(self, widget, event):
-
         if not hasattr(event, 'keyval') or event.keyval == 65293:
             self.win.remove_main_widget()
-            RegistrationScreen(self.win)
+            Gtk.main_quit()
```
Comment (skarbat) [inline kano_login/first_screen.py]: could we explain why we `Gtk.main_quit` and then `os.system`?

File: kano_login/first_screen.py [hunk 5]
```diff
@@ -93,18 +99,21 @@ def login_screen(self, widget, event):
...
+            os.system('{} {}{} {}'.format(FirstScreen.webengine, FirstScreen.web_api, FirstScreen.signup, \
+                                          FirstScreen.win_size))
 
     def exit_registration(self, widget, event):
         self.win.remove_main_widget()
+        sys.exit(0)
```
Comment (skarbat) [inline kano_login/first_screen.py]: likewise, I can't see the flow...

File: libs/kano-world/src/kano_world_functions.cpp [hunk 6]
```diff
@@ -72,3 +72,18 @@ std::string kano_world::functions::get_token() const
 
     return token;
 }
+
+void kano_world::functions::set_login_data(std::string id,
+std::string username, std::string email, std::string token) const
```
Comment (skarbat) [inline libs/kano-world/src/kano_world_functions.cpp]: Brief header explaining what we do below?

File: kano_login/first_screen.py (hunk 2 above)
Comment (tombettany) [reply kano_login/first_screen.py]: Methinks that we should move this to [`kano_world.config`](https://github.com/KanoComputing/kano-profile/blob/master/kano_world/config.py) and that way we can also set it through `/etc/kano-world.conf`

**Comment by tombettany** [comment]
@skarbat I've made some changes to address your comments. Not all of them as we need to get this shipped but enough that it will hopefully satisfy you.

**Comment by skarbat** [comment]
Thank you @tombettany - Let's go 👍

**Pull request merged**